nnsvs/mdn.py

```diff
@@ -0,0 +1,169 @@
+# coding: utf-8
+
+import torch
+from torch import nn
+import torch.nn.functional as F
+
+class MDNLayer(nn.Module):
+    """ Mixture Density Network layer
+
+    The input maps to the parameters of a Mixture of Gaussians (MoG) probability 
+    distribution, where each Gaussian has out_dim dimensions and diagonal covariance.
+
+    Implementation references:
+    1. Mixture Density Networks by Mike Dusenberry https://mikedusenberry.com/mixture-density-networks
+    2. PRML book https://www.microsoft.com/en-us/research/people/cmbishop/prml-book/
+    3. sagelywizard/pytorch-mdn https://github.com/sagelywizard/pytorch-mdn
+    4. sksq96/pytorch-mdn https://github.com/sksq96/pytorch-mdn
+
+    Arguments:
+        in_dim (int): the number of dimensions in the input
+        out_dim (int): the number of dimensions in the output
+        num_gaussians (int): the number of mixture component
+    Input:
+        minibatch (B, T, D_in): B is the batch size and T is data lengths of this batch, 
+            and D_in is in_dim.
+    Output:
+        log_pi, log_sigma, mu (B, T, G), (B, T, G, D_out), (B, T, G, D_out): 
+            G is num_gaussians and D_out is out_dim.
+            log_pi is log of a multinomial distribution of the Gaussians. 
+            mu and log_sigma are the mean and the log of standard deviation of each Gaussian.
+    """
```

Comment:
NOTE: this is my bad; I haven't mentioned the coding style explicitly, but I basically follow google's style guide https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html. 
For consistency, could you follow the guideline? e.g., Arguments -> Args.

Reply:
I changed the docstrings into google's style in the latest commit.